clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -157,14 +197,26 @@ object UncommittedGarbageCollector {
       firstSlice: String,
       startTime: java.time.Instant,
       success: Boolean,
+      expiredAddresses: DataFrame,
       removed: DataFrame
   ): Unit = {
-    val reportDst = s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
+    val reportDst = reportPath(storageNamespace, runID)
     writeJsonSummary(reportDst, runID, firstSlice, startTime, success, removed.count())
 
+    expiredAddresses.write.parquet(s"$reportDst/marked")
     removed.write.parquet(s"$reportDst/deleted")
   }
 
+  private def reportPath(storageNamespace: String, runID: String): String = {
+    s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
+  }
+
+  private def readMarkedAddresses(storageNamespace: String, markID: String): DataFrame = {
+    // TODO(eden) - check if this run succeeded
+    val path = reportPath(storageNamespace, markID)
+    spark.read.parquet(s"$path/marked")
+  }
+
   private def writeJsonSummary(
```

Comment:
How can we tell which run type was executed in this job? How can we verify it later?

Reply:
for now, I'm writing the JSON summary only for mark runs as we said we don't need the deletion info